docs/running-rasa-with-docker.rst

```diff
@@ -0,0 +1,437 @@
+
+.. code-block:: bash
+
+   docker run -v $(pwd):/app rasa/rasa init --no-prompt
+
+What does this command mean?
+
+- ``-v $(pwd):/app`` mounts your current working directory to the working directory
+  in the Docker container. This means that files you create on your computer will be
+  visible inside the container, and files created in the container will
+  get synced back to your computer.
+- ``rasa/rasa`` is the name of the docker image to run.
+- the Docker image has the ``rasa`` command as its entrypoint, which means you don't
+  have to type ``rasa init``, just ``init`` is enough.
+
+Running this command will produce a lot of output. What happens is:
+
+- a Rasa project is created (see :ref:`tutorial`)
+- an initial model is trained using the project's training data.
+
+To check that the command completed correctly, look at the contents of your working directory:
+
+.. code-block:: bash
+
+   ls -1
+
+The initial project files should all be there, as well as a ``models`` directory that contains your trained model.
+
+Talking to Your Assistant
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To talk to your newly-trained assistant, run this command:
+
+
+.. code-block:: bash
+
+   docker run -it -v $(pwd):/app rasa/rasa shell
+
+This will start a shell where you can chat to your assistant.
+Note that this command includes the flags ``-it``, which means that your are running
+Docker interactively, and you are able to give input via the command line.
+For commands which require interactive input, like ``rasa shell`` and ``rasa interactive``,
+you need to pass the ``-it`` flags.
+
+
+Customizing your Model
+----------------------
+
+Choosing a Tag
+~~~~~~~~~~~~~~
+
+To keep images as small as possible, we publish different tags of the ``rasa/rasa`` image
+with different dependencies installed. See :ref:`choosing_pipeline` for more information.
+
+All tags start with a version, the ``latest`` tag corresponds to the current master build.
+The tags are:
+
+- ``{version}``
+- ``{version}-spacy-en``
+- ``{version}-spacy-de``
+- ``{version}-mitie-en``
+- ``{version}-full``
+
+The plain ``latest`` tag includes all the dependencies you need to run the ``supervised_embeddings`` pipeline.
+If you are using components with pre-trained word vectors, you need to choose the corresponding tag.
+Alternatively, you can use the ``-full`` tag which includes all pipeline dependencies.
+
+.. note::
+
+   You can see a list of all the versions and tags of the Rasa Docker image
+   `here <https://hub.docker.com/r/rasa/rasa/>`_ .
+
+
+.. _model_training_docker:
+
+Training a Custom Rasa Model with Docker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Edit the ``config.yml`` file to use the pipeline you want, and place
+your NLU and Core data into the ``data/`` directory.
+Now you can train your own Rasa model by running:
+
+.. code-block:: bash
+
+  docker run \
+    -v $(pwd):/app \
+    rasa/rasa:latest-full \
+    train \
+      --domain domain.yml \
+      --data data/stories.md \
```

Comment:
@amn41 @erohmensing made a good point that `--stories` does not exist anymore. Do we want to give an example where we only provide `stories` or should we just use a directory?

Reply:
I think a dir is fine